backend/logic/printful/webhook.js

```diff
@@ -237,10 +253,42 @@ const processUpdatedEvent = async (event, shopId) => {
   )
 }
 
+/**
+ * Processes a ProductStockUpdated event sent by Printful via webhook.
+ * @param event
+ * @param shopId
+ * @param shopData
+ */
+const processStockUpdatedEvent = async (event, shopId, shopData) => {
+  try {
+    const { product_id } = event
+    const { shop, shopConfig } = shopData
+    const shopInventory = get(shopConfig, 'inventory')
+    if (shopInventory) {
+      const stockOutVariant = get(event, 'variant_stock.out', [])
+      const stockDiscontinuedVariant = get(
+        event,
+        'variant_stock.discontinued',
+        []
+      )
+
+      await stockUpdate(
+        shop,
+        product_id,
+        uniq([...stockOutVariant, ...stockDiscontinuedVariant])
```

Comment:
Should we test for this list to not be empty before calling stockUpdate?
Or handle that in the stockUpdate method

Reply:
It should be called even if the array is empty, but the `stockUpdate` function should update  `stockLeft` accordingly.

from printful Docs
> Contains product id and ids of it's discontinued variants and variants that are out of stock. Variant ids that are not present should be considered as active and in stock.

https://www.printful.com/docs/webhooks

Reply:
got it. thanks.